cpp/src/arrow/compute/kernels/cast_test.cc

```diff
@@ -604,6 +604,39 @@ TEST_F(TestCast, IntToFloatingPoint) {
 }
 #endif
 
+TEST_F(TestCast, DecimalToInt) {
+  CastOptions options;
+  options.allow_int_overflow = true;
+
+  std::vector<bool> is_valid = {true, true};
+  // simple cases decimal
+
+  std::vector<Decimal128> v = {Decimal128("02.0000000000"), Decimal128("30.0000000000")};
+  std::vector<int64_t> e = {2, 30};
+
+  CheckCase<Decimal128Type, Decimal128, Int64Type, int64_t>(decimal(38, 10), v, is_valid,
+                                                            int64(), e, options);
+}
+
+TEST_F(TestCast, DecimalToDecimal) {
+  CastOptions options;
+  options.allow_int_overflow = true;
+
+  std::vector<bool> is_valid = {true, true};
+  // simple cases decimal
```

Comment:
Same remark here: I would expect more tricky conversion cases, failures (overflow) and nulls.

Reply:
Added more tests: nulls, truncation